### PR TITLE
Add tests for 'WorkspaceFolderUpdater'

### PR DESCRIPTION
**Pull request by RomanNikitenko**
Signed-off-by: Roman Nikitenko <rnikiten@redhat.com>

### What does this PR do?
Add tests for 'WorkspaceFolderUpdater'

### What issues does this PR fix or reference?

It's related to multi-root workspaces support logic https://github.com/eclipse/che/issues/17212

### How to test this PR?

The tests should be happy.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: plugins/workspace-plugin/src/workspace-folder-updater.ts [hunk 1]
```diff
@@ -83,3 +83,7 @@ export class WorkspaceFolderUpdater {
     return path;
   }
 }
+
+export function setUpdateFolderTimeout(ms: number): void {
+  updateWorkspaceFolderTimeout = ms;
+}
```
[the remark this answers is not in the document]
Comment (benoitf) [reply plugins/workspace-plugin/src/workspace-folder-updater.ts:89]: What you can do is to use inversify binding to bind the constant and inject the value in the class
then in test you can bind the value you want

[the remark this answers is not in the document]
Comment (benoitf) [reply plugins/workspace-plugin/src/workspace-folder-updater.ts:89]: (and if you don't want to introduce inversify there, just provide value with constructor)

[the remark this answers is not in the document]
Comment (RomanNikitenko) [reply plugins/workspace-plugin/src/workspace-folder-updater.ts]: I applied the second option, thanks!

**Review by azatsarynnyy — APPROVED** [review]
:+1:

**Review by benoitf — APPROVED** [review]

**Pull request merged**